aws/resource_aws_cognito_user_pool_client.go

```diff
@@ -0,0 +1,215 @@
+package aws
+
+import (
+	"log"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/service/cognitoidentityprovider"
+	"github.com/hashicorp/errwrap"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceAwsCognitoUserPoolClient() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsCognitoUserPoolClientCreate,
+		Read:   resourceAwsCognitoUserPoolClientRead,
+		Update: resourceAwsCognitoUserPoolClientUpdate,
+		Delete: resourceAwsCognitoUserPoolClientDelete,
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"generate_secret": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				ForceNew: true,
+			},
+
+			"user_pool_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"explicit_auth_flows": {
+				Type:     schema.TypeList,
+				Optional: true,
+				ForceNew: false,
+				Elem: &schema.Schema{
+					Type:         schema.TypeString,
+					ValidateFunc: validateCognitoUserPoolClientAuthFlows,
+				},
+			},
+
+			"read_attributes": {
+				Type:     schema.TypeList,
+				Optional: true,
+				ForceNew: false,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+
+			"write_attributes": {
+				Type:     schema.TypeList,
+				Optional: true,
+				ForceNew: false,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+
+			"refresh_token_validity": {
+				Type:         schema.TypeInt,
+				Optional:     true,
```

Comment:
The default validity seems to be set to `30` by the API which is why the acceptance test is failing:

```
=== RUN   TestAccAWSCognitoUserPoolClient_basic
--- FAIL: TestAccAWSCognitoUserPoolClient_basic (34.36s)
	testing.go:503: Step 0 error: After applying this step, the plan was not empty:

		DIFF:

		UPDATE: aws_cognito_user_pool_client.client
		  refresh_token_validity: "30" => "0"
```

Do you mind setting `Default: 30` here too?

Reply:
No issues with that